tinylog-impl/src/main/java/org/tinylog/impl/path/segments/DateTimeSegment.java

```diff
@@ -59,6 +59,10 @@ public DateTimeSegment(String pattern, Locale locale) {
 
 	@Override
 	public String findLatest(Path parentDirectory, String prefix) throws IOException {
+		if (!Files.exists(parentDirectory)) {
+			return null;
+		}
```

Comment:
Thank you for the fix. I have noticed that we have the same issue in `CountSegment` (and probably other future path segments). What do you think about fixing the issue for all segments in [org/tinylog/impl/path/DynamicPath.java#L40](https://github.com/tinylog-org/tinylog/blob/8f7ffbd34bc45e1fe09c696aad89af041ce9bd92/tinylog-impl/src/main/java/org/tinylog/impl/path/DynamicPath.java#L40)? You would just have to move the new if statement there.

Reply:
Yeah of course. I will do it immediately

Reply:
I think it should be in the loop after `for (int i = 0; i < pathSegments.length; ++i) {` and before `String result = pathSegments[i].findLatest(parentDirectory, prefix);`. So, your check will be executed before calling each path segment.

Could you update the JUnit test, too?

Reply:
I moved it inside the loop and wrote a unit test.